assets/js/atomic/blocks/product-elements/price/edit.js

```diff
@@ -31,7 +39,7 @@ const PriceEdit = ( { attributes, setAttributes } ) => {
 				) }
 			</BlockControls>
 			<div { ...blockProps }>
-				<Block { ...attributes } />
+				<Block { ...{ ...attributes, ...context } } />
```

Comment:
I'm not 100% sure I get this piece of code. Why does this need to take this shape and why is this spreading required? 🤔

Reply:
I find it a bit hard to read, honestly. Perhaps it's because it's not TS and I am not sure what the Block is expecting here.

My suggestion is that we either go with the “non-shorthand” version (which, funnily enough, even has 4 characters less than the shorthand), or we define this object somewhere before and then spread it.

```jsx
const blockAttrs = {
   ...attributes,
   ...context,
};

return (
  <div { ...blockProps }>
    <Block { ...blockAttrs } />
  </div>
);
```

But the non-shorthand version looks nicer to me. What do you think?

Reply:
> Perhaps it's because it's not TS and I am not sure what the Block is expecting here.

I wonder if converting this to TS will resolve this. In either case, I'm good with changing this up as long as we can be consistent elsewhere in our codebase. Between the two options, I actually like the abstracted constant in your example above!

Reply:
Would definitely help. That's what I find often super hard with working with Gutenberg. I never know the shape that a block is expecting (and in fact the changes we requested even broke types with the latest addition of the `allowedControls`, and they didn't seem to mind too much).

With that said, I feel that the repeated spreading within repeated curly braces is still quite confusing and requires me at least a moment to parse. I prefer to optimize for cognitive ease most of the time (considering this doesn't even optimize for length of code, I think it's a win-win either way).

Reply:
I agree. I like the idea of favoring readability (hence maintainability). I happen to have seen this syntax before so I didn't think too much of it. But you're certainly right about the point that it takes an extra min to reason about and I'm more than happy to make this adjustment.

> which, funnily enough, even has 4 characters less than the shorthand

Haha, true. Shorthand may not be the proper word here. Condensed or merged is probably more accurate. 😜

Reply:
Updated in f64a083.